recipes/brunsli/use_conda_forge_shared_libraries.patch

```diff
@@ -0,0 +1,96 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index 7d785be..edcf192 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -26,7 +26,7 @@ set(CMAKE_CXX_STANDARD 11)
+ set(CMAKE_CXX_EXTENSIONS OFF)
+ set(CMAKE_CXX_STANDARD_REQUIRED YES)
+ 
+-add_subdirectory(third_party)
++# add_subdirectory(third_party)
+ 
+ # The Brunsli library definition.
+ include(brunsli.cmake)
+diff --git a/brunsli.cmake b/brunsli.cmake
+index 1e87527..cd2428f 100644
+--- a/brunsli.cmake
++++ b/brunsli.cmake
+@@ -57,7 +57,8 @@ add_library(brunslidec-static STATIC
+   ${BRUNSLI_DEC_HEADERS}
+ )
+ target_link_libraries(brunslidec-static PRIVATE
+-  brotlidec-static
++  brotlidec
++  brotlicommon
+   brunslicommon-static
+ )
+ 
+@@ -66,7 +67,8 @@ add_library(brunslienc-static STATIC
+   ${BRUNSLI_ENC_HEADERS}
+ )
+ target_link_libraries(brunslienc-static PRIVATE
+-  brotlienc-static
++  brotlienc
++  brotlicommon
+   brunslicommon-static
+ )
+ 
+@@ -80,7 +82,16 @@ target_link_libraries(brunslidec-c PRIVATE brunslidec-static)
+ add_library(brunslienc-c SHARED
+   c/enc/encode.cc
+ )
++target_link_libraries(brunslidec-c PRIVATE
++  brotlidec
++  brotlicommon
++)
++
+ target_link_libraries(brunslienc-c PRIVATE brunslienc-static)
++target_link_libraries(brunslienc-c PRIVATE
++  brotlienc
++  brotlicommon
++)
+ list(APPEND BRUNSLI_LIBRARIES brunslidec-c brunslienc-c)
+ endif()  # BRUNSLI_EMSCRIPTEN
+ 
+@@ -98,10 +109,14 @@ if(NOT BRUNSLI_EMSCRIPTEN)
+ add_executable(cbrunsli c/tools/cbrunsli.cc)
+ target_link_libraries(cbrunsli PRIVATE
+   brunslienc-static
++  brotlienc
++  brotlicommon
+ )
+ add_executable(dbrunsli c/tools/dbrunsli.cc)
+ target_link_libraries(dbrunsli PRIVATE
+   brunslidec-static
++  brotlidec
++  brotlicommon
+ )
+ else()  # BRUNSLI_EMSCRIPTEN
+ set(WASM_MODULES brunslicodec-wasm brunslidec-wasm brunslienc-wasm)
+@@ -133,11 +148,24 @@ endif()  # BRUNSLI_EMSCRIPTEN
+ # Installation
+ if(NOT BRUNSLI_EMSCRIPTEN)
+   install(
+-    TARGETS brunslidec-c brunslienc-c
+-    ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}"
++    TARGETS brunslidec-c
++    RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}"
++    ARCHIVE DESTINATION "${CMAKE_INSTALL_BINDIR}"
```

Comment:
```suggestion
+    ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}"
```

Reply:
I diagree with this suggestion
```
ARCHIVE

    Target artifacts of this kind include:

        Static libraries (except on macOS when marked as FRAMEWORK, see below);

        DLL import libraries (on all Windows-based systems including Cygwin; they have extension .lib, in contrast to the .dll libraries that go to RUNTIME);
```

Reply:
The defaults are sane for each platform, so I generally don't specify these unless I really wish to override. I would recommend watching this excellent talk: https://www.youtube.com/watch?v=m0DwB4OvDXk (can skip to 28:02 for this specific topic)